### DISCUSSION TITLE
Rest API chunk upload

### DISCUSSION THREAD

**Discussion by MohammadBMelhem** [dadoonet/fscrawler#1572] [Q&A]
Hello,

I want to use it as rest API on docker, but I have an issue, what if I have a big file of 500MB and I need to index the file as chunks I don't want to load the full file on the memory, any suggestion? thanks,

another question: any reason to index the content as filed and not uses the attachment on elastic?

Thanks.

**Comment by dadoonet** [comment]
>  I need to index the file as chunks I don't want to load the full file on the memory, any suggestion? thanks,

we don't support this in FSCrawler.

> any reason to index the content as filed and not uses the attachment on elastic?

You can use the ingest attachment processor if you wish but that'd require to send the full content as a json to Elasticsearch and may be overload a bit the node.
I found better to extract the meaningful data where the data lives and then send only that to Elasticsearch.
Also FSCrawler supports OCR and many other file types than the ingest attachment processor.
